packages/utils/src/logger/logger.ts

```diff
@@ -37,55 +37,53 @@ export abstract class Logger {
   abstract info(message: unknown, ...args: unknown[]): void
   abstract debug(message: unknown, ...args: unknown[]): void
 }
-class DebugLogger extends Logger {
+class DebugLogger implements Logger {
   private readonly logger: Debug.Debugger
 
   public constructor(category?: string) {
-    super()
     this.logger = Debug(category)
   }
 
   protected log(message: unknown, ...args: unknown[]): void {
-    if (args?.length) this.logger.log(message, args)
-    else this.logger.log(message)
+    if (args?.length) this.logger(message, ...args)
+    else this.logger(message)
   }
 
   public debug(message: unknown, ...args: unknown[]): void {
     this.logger.log = console.debug.bind(console)
-    this.log(message, args)
+    this.log(message, ...args)
   }
 
   public error(message: unknown, ...args: unknown[]): void {
     this.logger.log = console.error.bind(console)
-    this.log(message, args)
+    this.log(message, ...args)
   }
 
   public info(message: unknown, ...args: unknown[]): void {
     this.logger.log = console.log.bind(console)
-    this.log(message, args)
+    this.log(message, ...args)
   }
 
   public warn(message: unknown, ...args: unknown[]): void {
     this.logger.log = console.warn.bind(console)
-    this.log(message, args)
+    this.log(message, ...args)
   }
 
   public fatal(message: unknown, ...args: unknown[]): void {
     this.logger.log = console.error.bind(console)
-    this.log(message, args)
+    this.log(message, ...args)
   }
 
   public trace(message: unknown, ...args: unknown[]): void {
     this.logger.log = console.trace.bind(console)
-    this.log(message, args)
+    this.log(message, ...args)
   }
 }
 
-class ConfigLogger extends Logger {
+class ConfigLogger implements Logger {
```

Comment:
Didn't you could implement an abstract class oO
Confusing for me coming from Java for OOP-related stuff. Any TS-specific resource to understand this?

Reply:
I find myself reading https://www.freecodecamp.org/news/advanced-typescript-types-cheat-sheet-with-examples/ multiple times per week...